test/test-define-x-expander-use-site-scope.rkt

```diff
@@ -0,0 +1,73 @@
+#lang racket
+
+(require syntax/parse
+         syntax/parse/experimental/eh
+         generic-syntax-expanders
+         syntax/stx
+         (for-syntax syntax/parse
+                     racket/syntax)
+         rackunit)
+
+(define-expander-type eh-mixin)
+
+(begin-for-syntax
+  (define eh-post-accumulate (make-parameter #f)))
+
+(define-for-syntax (strip-use-site stx)
+  (define bd
+    (syntax-local-identifier-as-binding (syntax-local-introduce #'here)))
+  (define delta
+    (make-syntax-delta-introducer (syntax-local-introduce #'here) bd))
+  (delta stx 'remove))
+
+(define-syntax define-eh-alternative-mixin
+  (syntax-parser
+    [(_ name ((~literal pattern) pat) ... (~optional (~seq #:post post)))
+     (let ()
+       #`(define-eh-mixin-expander name
+           (λ (_)
+             #,@(if (attribute post)
+                    #`(((eh-post-accumulate) (quote-syntax post)))
+                    #'())
+             (quote-syntax (~or . #,(strip-use-site #'(pat ...)))))))]))
+
+(define-syntax ~no-order
+  (pattern-expander
+   (λ (stx)
+     (syntax-case stx ()
+       [(self pat ...)
```

Comment:
Does this work if `pat` is a keyword? If not, I'd recommend using `syntax-parser` and the `:expr` syntax class, which disallows keywords.